src/main/java/net/nextencia/rrdiagram/BNFProcessor.java

```
@@ -0,0 +1,262 @@
+    try {
+      FileReader in = new java.io.FileReader(bnffile);
+      BNFToGrammar btg = new BNFToGrammar();
+      grammar = btg.convert(in);
+      logger.info("loaded: " + bnffile);
+      this.api = api;
+      this.version = version;
+    } catch (java.io.FileNotFoundException fne) {
+      logger.severe("File not found: " + bnffile);
+      System.exit(1);
+    } catch (java.io.IOException ioe) {
+      logger.severe("Unable to read from file: " + bnffile);
+      System.exit(1);
+    }
+  }
+
+  public String getReferenceFile() {
+    StringBuilder sb = new StringBuilder();
+
+    try {
+      checkGrammar();
+
+      GrammarToBNF bnf_builder = new GrammarToBNF();
+      GrammarToRRDiagram.RuleLinkProvider ruleLinkProvider =
+        new GrammarToRRDiagram.RuleLinkProvider() {
+          private String toLinkName(String name) {
+            return "../grammar_diagrams#" + name.replaceAll("_", "-");
+          }
+
+          @Override
+          public String getLink(String ruleName) {
+            return toLinkName(ruleName);
+          }
+        };
+
+      for (Rule rule : grammar.getRules()) {
+        try {
+          sb.append("### " + rule.getName() + "\n");
+
+          sb.append("```output.ebnf\n");
+          sb.append(rule.toYBNF());
+          sb.append("\n```\n");
+          GrammarToRRDiagram diagram_builder = new GrammarToRRDiagram();
+          diagram_builder.setRuleLinkProvider(ruleLinkProvider);
+          diagram_builder.setRuleConsideredAsLineBreak(Utils.lineBreakRule);
+
+          RRDiagram diagram = diagram_builder.convert(rule);
+          String svg_string = new RRDiagramToSVG().convert(diagram);
+          sb.append(svg_string);
+          sb.append("\n\n");
+        } catch (Exception e) {
+          logger.severe("Exception occurred while exporting rule " + rule.getName());
+          logger.warning(rule.toBNF(bnf_builder));
+          throw e;
+        }
+      }
+    } catch(Exception e) {
+      e.printStackTrace();
+    }
```

why is exception ignored? do you expect it to happen?

I'm seeing this often.  Are you intentionally doing this for each case or defaulting to this practice?

---

`printstacktrace` is more for debugging as I'm unfamiliar with all the exceptions that the RRDiagram code can throw. Are you thinking of not handling the exception but throwing it back to the caller?

---

It is bad practice to silently ignore errors because it can lead to hidden behavior.  I'd rather things crash than serve incorrect/corrupt information.